### PR TITLE
Pro 6694 breakpoint preview vite ready

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-6694](https://linear.app/apostrophecms/issue/PRO-6694/breakpoint-preview-css-loader-logic-should-be-vite-ready)

## Summary

Uses new externalize postcss plugin instead of having one postcss plugin and one webpack loader.

Cypress tests running [HERE](https://github.com/apostrophecms/testbed/actions/runs/11775851651) 🟢 

## What are the specific steps to test this change?

Should be tested with this [new plugin](https://github.com/apostrophecms/postcss-viewport-to-container-toggle/pull/1):

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/asset/lib/webpack/src/webpack.scss.js [hunk 1]
```diff
@@ -1,26 +1,16 @@
-const path = require('path');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
-const postcssReplaceViewportUnitsPlugin = require('../postcss-replace-viewport-units-plugin');
+const postcssViewportToContainerToggle = require('postcss-viewport-to-container-toggle');
 
 module.exports = (options, apos, srcBuildNames) => {
   const postcssPlugins = [
+    ...apos.asset.options.breakpointPreviewMode?.enable === true ? [
+      postcssViewportToContainerToggle({
+        modifierAttr: 'data-breakpoint-preview-mode'
```
Comment (haroun) [inline modules/@apostrophecms/asset/lib/webpack/src/webpack.scss.js:8]: please pass transform and debug to the plugin

File: modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js [hunk 2]
```diff
@@ -1,42 +1,31 @@
-const path = require('path');
-const postcssReplaceViewportUnitsPlugin = require('../postcss-replace-viewport-units-plugin');
+const postcssViewportToContainerToggle = require('postcss-viewport-to-container-toggle');
 
 module.exports = (options, apos) => {
   const postcssPlugins = [
+    ...apos.asset.options.breakpointPreviewMode?.enable === true ? [
+      postcssViewportToContainerToggle({
+        modifierAttr: 'data-breakpoint-preview-mode'
```
Comment (haroun) [inline modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js:7]: please pass transform and debug to the plugin

File: modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js [hunk 3]
```diff
@@ -1,42 +1,31 @@
...
-        debug: apos.asset.options.breakpointPreviewMode?.debug === true,
-        transform: apos.asset.options.breakpointPreviewMode?.transform || null
-      }
-    };
-  }
 
   return {
     module: {
       rules: [
+        // TODO: Why no postcss plugin here while loader was here?
```
Comment (haroun) [inline modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js:17]: Should be tested in pure css files

**Review by haroun — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js (hunk 3 above)
Comment (ValJed) [reply modules/@apostrophecms/asset/lib/webpack/apos/webpack.scss.js]: postcss added to css rule.

File: modules/@apostrophecms/asset/lib/webpack/src/webpack.scss.js (hunk 1 above)
Comment (ValJed) [reply modules/@apostrophecms/asset/lib/webpack/src/webpack.scss.js]: done

File: modules/@apostrophecms/asset/index.js [hunk 4]
```diff
@@ -494,7 +494,13 @@ module.exports = {
           isTask: !argv['check-apos-build'],
           hmr: self.hasHMR(),
           hmrPort: self.options.hmrPort,
-          sourcemaps: self.options.productionSourceMaps
+          sourcemaps: self.options.productionSourceMaps,
+          postcssViewportToContainerToggle: {
```
Comment (myovchev) [inline modules/@apostrophecms/asset/index.js:498]: Please document the new options to the dev comments above - what every prop does mean and represents.

**Review by myovchev — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/asset/index.js (hunk 4 above)
Comment (ValJed) [reply modules/@apostrophecms/asset/index.js:499]: Right, added a line for `postcssViewportToContainerToggle`

File: modules/@apostrophecms/asset/index.js [hunk 5]
```diff
@@ -494,7 +495,13 @@ module.exports = {
...
+            enable: self.options.breakpointPreviewMode?.enable,
```
Comment (ETLaurent) [inline modules/@apostrophecms/asset/index.js:500]:
```suggestion
            enable: self.options.breakpointPreviewMode?.enable === true,
```

like below?

**Review by ETLaurent — APPROVED** [review]
lgtm

changelog entry?

File: modules/@apostrophecms/asset/index.js (hunk 5 above)
Comment (ValJed) [reply modules/@apostrophecms/asset/index.js:500]: Yes!

**Review by haroun — APPROVED** [review]

**Review by myovchev — APPROVED** [review]

**Review by myovchev — APPROVED** [review]

**Pull request merged**